packages/job-worker/src/playout/abPlayback/index.ts

```diff
@@ -6,35 +6,45 @@ import { WrappedShowStyleBlueprint } from '../../blueprints/cache'
 import { ReadonlyDeep } from 'type-fest'
 import { JobContext, ProcessedShowStyleCompound } from '../../jobs'
 import { getCurrentTime } from '../../lib'
-import { resolveAbAssignmentsFromRequests } from './abPlaybackResolver'
+import { resolveAbAssignmentsFromRequests, SessionRequest } from './abPlaybackResolver'
 import { calculateSessionTimeRanges } from './abPlaybackSessions'
 import { applyAbPlayerObjectAssignments } from './applyAssignments'
 import { AbSessionHelper } from './abSessionHelper'
 import { ShowStyleContext } from '../../blueprints/context'
 import { logger } from '../../logging'
 import { ABPlayerDefinition } from '@sofie-automation/blueprints-integration'
+import { PlayoutModel } from '../model/PlayoutModel'
+import { applyAndValidateOverrides } from '@sofie-automation/corelib/dist/settings/objectWithOverrides'
+import { StudioRouteSet } from '@sofie-automation/shared-lib/dist/core/model/StudioRouteSet'
+
+interface MembersOfRouteSets {
+	poolName: string
+	playerId: string | number
+	disabled: boolean
+}
 
 /**
  * Resolve and apply AB-playback for the given timeline
  * @param context Context of the job
  * @param abSessionHelper Helper for generation sessionId
  * @param blueprint Blueprint of the currently playing ShowStyle
  * @param showStyle The currently playing ShowStyle
- * @param playlist The currently playing Playlist
+ * @param playoutModel The current playout model
  * @param resolvedPieces All the PieceInstances on the timeline, resolved to have 'accurate' playback timings
  * @param timelineObjects The current timeline
  * @returns New AB assignments to be persisted on the playlist for the next call
  */
-export function applyAbPlaybackForTimeline(
+export async function applyAbPlaybackForTimeline(
 	context: JobContext,
 	abSessionHelper: AbSessionHelper,
 	blueprint: ReadonlyDeep<WrappedShowStyleBlueprint>,
 	showStyle: ReadonlyDeep<ProcessedShowStyleCompound>,
-	playlist: ReadonlyDeep<DBRundownPlaylist>,
+	playoutModel: PlayoutModel,
 	resolvedPieces: ResolvedPieceInstance[],
 	timelineObjects: OnGenerateTimelineObjExt[]
-): Record<string, ABSessionAssignments> {
+): Promise<Record<string, ABSessionAssignments>> {
 	if (!blueprint.blueprint.getAbResolverConfiguration) return {}
+	const playlist = playoutModel.playlist as DBRundownPlaylist
```

Comment:
This cast is a bad idea. If you are having problems with it being `ReadonlyDeep<...>`, then that might need propogating through.
This type is being used everywhere to make it clear in the code that it must not be modified. By discarding that, it is not clear in this code if it is safe/allowed to modify various values. And doing so could cause hard to diagnose bugs, as the model will not fully understand what has changed.